app/src/main/res/layout/activity_client_registration.xml

```diff
@@ -1,173 +1,185 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="@color/teal_700"
-    tools:context=".frontend.ClientRegistrationActivity">
-    <ScrollView
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <data>
+        <variable
+            name="cdata"
+            type="com.javahelp.frontend.ClientRegistrationVm" />
+
+    </data>
+
```

Comment:
What is this part doing? I am not familiar with this.

Reply:
I think you can understand this as you create a ViewModel and you linked the data in this XML to ViewModel and then ViewModel linked to the binding